http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/AwsCrtConfigurationUtils.java

```diff
@@ -41,6 +41,7 @@ public static SocketOptions buildSocketOptions(TcpKeepAliveConfiguration tcpKeep
         }
 
         if (tcpKeepAliveConfiguration != null) {
+            clientSocketOptions.keepAlive = true;
```

Comment:
Can we add a test for this? Can we also add a changelog entry? It can be done by running the scripts/new-change script and following the instructions. Commit the new file created by the script in .changes/next-release with your changes.

Reply:
Thank you for your quick reply.
I think this PR might be obsoleted by https://github.com/awslabs/aws-crt-java/pull/751 ?

Reply:
https://github.com/awslabs/aws-crt-java/pull/751 was a proposal of a more user-friendly API on setting keep alive, that's why it was sitting there unmerged. I'm going to merge it after pulling last changes from the main.

Reply:
Thanks @sfod 
@eckardnet it seems we still need to make change to either set ` clientSocketOptions.keepAlive` to true or use the new API `setTcpKeepAlive(keepAliveTimeoutSecs, keepAliveIntervalSecs)` here.